packages/core/test/lib/console-child.js

```diff
@@ -0,0 +1,86 @@
+const assert = require("chai").assert;
+const fs = require("fs-extra");
+const path = require("path");
+const tmp = require("tmp");
+const TruffleConfig = require("@truffle/config");
+const { deriveConfigEnvironment } = require("../../lib/command-utils");
+
+let config;
+
+function createSandbox(source) {
+  if (!fs.existsSync(source)) {
+    throw new Error(`Sandbox failed: source: ${source} does not exist`);
+  }
+
+  const tempDir = tmp.dirSync({ unsafeCleanup: true });
+  fs.copySync(source, tempDir.name);
+  const config = TruffleConfig.load(
+    path.join(tempDir.name, "truffle-config.js"),
+    {}
+  );
+  return config;
+}
+
+describe("console-child", function () {
+  before("Create a sandbox", function () {
+    config = createSandbox(
+      path.join(__dirname, "..", "sources", "console-child")
+    );
+  });
+
+  describe("deriveConfigEnvironment", function () {
+    it("network has provider", function () {
+      config.network = "crazyTimeNetwork";
+      const expectedNetworkConfig = config.networks[config.network];
+      const cfg = deriveConfigEnvironment(config, config.network, "");
+      assert.equal(cfg.network, config.network);
+      assert.equal(
+        cfg.networks[config.network].confirmations,
+        expectedNetworkConfig.confirmations
+      );
+    });
+
+    it("url is specified instead of network", function () {
+      config.network = "localhost:5555";
```

Comment:
I kind of feel like you should change this network name so it doesn't match the url provivded as a third argument. It seems it is likely to confuse people in that they might think there is some relationship between the third argument provided and this network name when, in fact, they are totally unrelated.

Reply:
When we specify the `--url` option with `truffle console` command, then the `network` name happens to be the `host` of the url. It is the logic that goes behind `mergeConfigNetwork.js` inside the `core/lib` package.

Reply:
Yeah I understand, but the network name in this case is arbitrary and has no bearing on the results (at least in this function). In other words, you could name it `"sukanyaNetwork"` and this test will correctly pass. I'm suggesting you change it to something like that so people won't think it has something to do with the `url` parameter.

Reply:
I got your point @eggplantzzz. The network name doesn't matter here much. I will apply your suggestions :)